ciao-controller/internal/datastore/sqlite3db.go

```diff
@@ -2305,7 +2306,14 @@ func (ds *sqliteDB) deleteBlockData(ID string) error {
 
 func (ds *sqliteDB) createStorageAttachment(a types.StorageAttachment) error {
 	ds.dbLock.Lock()
-	err := ds.create("attachments", a.ID, a.InstanceID, a.BlockID)
+	var ephemeral int
+
+	if a.Ephemeral {
+		ephemeral = 1
+	} else {
+		ephemeral = 0
+	}
+	err := ds.create("attachments", a.ID, a.InstanceID, a.BlockID, ephemeral)
 	ds.dbLock.Unlock()
```

Comment:
I don't think you need to convert your bool to an integer for sqlite3 - I believe it translates that for you (from a bool to an int).